### PR TITLE
Configure lint continuous integration

### PR DESCRIPTION
**Pull request by namoscato**
Resolves https://github.com/cedpoilly/vue-gapi/issues/58.

## Details

* Cleanup ESLint configuration
  * merge `.eslintrc.js` into `.eslintrc.json`
  * upgrade ESLint and friends
  * add [`eslint-config-prettier`](https://github.com/prettier/eslint-config-prettier)
* Operationalize Prettier
* Remove a few additional CSS things, missed in https://github.com/cedpoilly/vue-gapi/pull/64
* Add `babel-jest` in favor of `babel-plugin-istanbul`, i.e. see [Using Babel](https://jestjs.io/docs/en/getting-started#using-babel)
* Upgrade `lint-staged`

## Dependencies

* https://github.com/cedpoilly/vue-gapi/pull/64

## Checklist

- [x] fix ESLint errors
- [x] fix Prettier errors
- [x] determine what to do with [lint-staged](https://github.com/okonet/lint-staged)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
   },
-  "env": {
-    "browser": true,
-    "node": true
+  env: {
+    browser: true,
+    commonjs: true,
+    es6: true,
   },
 }
```
Comment (namoscato) [inline .eslintrc.js:18]: Open to suggestions on this configuration ☝️ 

`.eslintrc.json` would never have been relevant as `.eslintrc.js` takes precedence per [Configuration File Formats](https://eslint.org/docs/user-guide/configuring#configuration-file-formats); thus, I just merged the former into here.

I'm not sure it makes sense to extend `plugin:vue/recommended`, for example, given that this project doesn't really have any `.vue` files, but I left it in here for closer compatibility.

Comment (cedpoilly) [reply .eslintrc.js:18]: Good for eslint files

For config related to vue, let's keep it for now ;)

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1 @@
+dist
```
Comment (namoscato) [inline .prettierignore:1]: _See [Ignoring Files: .prettierignore](https://prettier.io/docs/en/ignore.html#ignoring-files-prettierignore)._

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -19,4 +19,20 @@ jobs:
...
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+      - name: Install dependencies
+        run: npm ci
+      - name: Check code style
+        run: npm run lint
```
Comment (namoscato) [inline .github/workflows/ci.yaml:38]: _This configures a second [job](https://docs.github.com/en/actions/getting-started-with-github-actions/core-concepts-for-github-actions#job) for the `ci` workflow (defined by this YAML file)._

_After checking out code, this reads the support Node.js version as defined by `.nvmrc` and invokes a [workflow command](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-output-parameter) to setup Node.js._

File: build/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "env": {
+    "node": true
+  }
+}
```
Comment (namoscato) [inline build/.eslintrc.json:5]: _This extends the ESLint configuration defined in the root, only adding the Node.js environment to build-related files. See [Configuration Cascading and Hierarchy](https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy) for more information._

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (cedpoilly) [reply .github/workflows/ci.yaml:38]: Noted

File: .prettierignore (hunk 2 above)
Comment (cedpoilly) [reply .prettierignore:1]: Good

**Comment by namoscato** [comment]
This is good to go!

**Pull request merged**